### PR TITLE
balloons: PreferSpreadOnPhysicalCores & balloon-specific CPU allocator

### PR DESCRIPTION
**Pull request by askervin**

### CODE REVIEW

File: docs/policy/balloons.md [hunk 1]
```diff
@@ -135,6 +142,8 @@ Balloons policy parameters:
     - `numa`: ...in the same numa node(s) as the balloon.
     - `core`: ...allowed to use idle CPU threads in the same cores with
       the balloon.
+  - `PreferManyCores` overrides the policy level option with the same
```
Comment (kad) [inline docs/policy/balloons.md:145]: Many->More?

File: docs/policy/balloons.md [hunk 2]
```diff
@@ -85,6 +85,13 @@ Balloons policy parameters:
   pack new balloons tightly into the same NUMAs/dies/packages. This
   helps keeping large portions of hardware idle and entering into deep
   power saving states.
+- `PreferMoreCores` prefers allocating logical CPUs for a balloon from
```
Comment (kad) [inline docs/policy/balloons.md:88]: Maybe somehow name to reference physical vs hyperthread "cores"?

Comment (askervin) [reply docs/policy/balloons.md:88]: `PreferSpreadOnPhysicalCores`

**Review by marquiz — APPROVED** [review]
Looks good. Although I must admit I didn't think through all the details of the algorithm 😎

LGTM

**Pull request merged**